docs/examples/01_current_injection.py

```diff
@@ -1,502 +1,694 @@
-# $$ \lambda (t) = \exp (w \cdot i(t) + c) \tag{1}$$
+#
+# Our last step is to convert these to `jax.numpy` arrays.
 
-# apply an exponential non-linearity to L(t)
-predicted_rate = np.exp(L)
+predictor = jax.numpy.asarray(predictor.values)
+count = jax.numpy.asarray(count.values)
 
 # %%
-# The same linear-non-linear transformation is implemented by the `predict` method of the nemos.glm.GLM
-# object.
+#
+# !!! info
+#
+#     In this example, we're being very explicit about this conversion to
+#     jax.numpy arrays. However, in general, nemos is able to properly convert
+#     from pynapple objects to jax.numpy arrays without any additional steps
+#     (it can similarly convert from numpy arrays to jax.numpy arrays). Thus,
+#     in later tutorials we will omit this step.
+#
+# ### Fitting the model
+#
+# Now we're ready to fit our model!
+#
+# First, we need to define our GLM model object. We intend for our models to be
+# similar to
+# [scikit-learn's](https://scikit-learn.org/stable/getting_started.html)
+# estimators: they are initialized with hyperparameters that specify
+# optimization and model details, and then the user calls `.fit` with the
+# model matrix and data.
+#
+# To initialize our model, we need to specify the regularizer and observation
+# model objects, both of which should be one of our custom objects:
+#
+# - Regularizer: this object specifies both the solver algorithm and the
+#   regularization scheme. They are jointly specified because each
+#   regularization scheme has a list of compatible solvers to choose between.
+#   Regularization modifies the objective function to reflect your prior
+#   beliefs about the parameters, such as sparsity. Regularization becomes more
+#   important as the number of input features, and thus model parameters,
+#   grows. They can be found within `nemos.regularizer`.
+#
+# !!! warning
+#
+#     With a convex problem like the GLM, in theory it does not matter which
+#     solver algorithm you use. In practice, due to numerical issues, it might.
+#     Thus, it's worth trying a couple to see how their solutions compare.
+#     (Different regularization schemes will always give different results.)
+#
+# - Observation model: this object links the firing rate and the observed
+#   spikes, describing the distribution of neural activity (and thus changing
+#   the log-likelihood). For now, the only possible observation model is the
+#   Poisson, though this may change in future releases. They can be found
+#   within `nemos.observation_models`.
+#
+# !!! warning
+#     how's this description?
+#
+# For this example, we'll use an un-regularized LBFGS solver. We'll discuss
+# regularization in a later tutorial.
+#
+# !!! info "Why LBFGS?"
+#
+#     LBFGS is a second-order optimizer, that is, it uses both the first
+#     derivative (the gradient) and the second derivative in order to solve the
+#     problem. This leads to a faster solution. Try other solvers to see how
+#     they behave!
```

Comment:
This is incorrect, LBFGS only uses gradients but it tries to approximate the Hessian by storing a recent history of gradients. It never explicitly computes a second derivative so it is called a "quasi-Newton method."

It would be good to add a link to the wikipedia page: https://en.wikipedia.org/wiki/Limited-memory_BFGS

For some of these models a true Newton's method should work very well. Have we played with this?

Fun aside: the "G" in LBFGS stands for Goldfarb and I guess he hangs around CCM. I randomly had lunch with him a couple weeks ago.

Reply:
How's the change I made?